controllers/inferenceservice_route.go

```diff
@@ -93,25 +96,73 @@ func CompareInferenceServiceRoutes(r1 routev1.Route, r2 routev1.Route) bool {
 		reflect.DeepEqual(r1.Spec, r2.Spec)
 }
 
+func (r *OpenshiftInferenceServiceReconciler) findSupportingRuntimeForISvc(ctx context.Context, log logr.Logger, inferenceservice *inferenceservicev1.InferenceService) *predictorv1.ServingRuntime {
+	desiredServingRuntime := predictorv1.ServingRuntime{}
+
+	if inferenceservice.Spec.Predictor.Model.Runtime != nil {
+		err := r.Get(ctx, types.NamespacedName{
+			Name:      *inferenceservice.Spec.Predictor.Model.Runtime,
+			Namespace: inferenceservice.Namespace,
+		}, &desiredServingRuntime)
+		if err != nil {
+			if apierrs.IsNotFound(err) {
+				log.Info("Runtime specified in InferenceService does not exist", "runtime", *inferenceservice.Spec.Predictor.Model.Runtime)
+			}
+		}
+		return &desiredServingRuntime
+	} else {
+		runtimes := &predictorv1.ServingRuntimeList{}
+		err := r.List(ctx, runtimes, client.InNamespace(inferenceservice.Namespace))
+		if err != nil {
+			return nil
```

Comment:
Could you add a log.Info so it shows why it errored out

Reply:
@heyselbi Done!